dev-tools/xtask/src/a4x2/a4x2_package.rs

```diff
@@ -0,0 +1,399 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Subcommand: cargo xtask a4x2-package
+
+use super::cmd;
+use anyhow::{Context, Result};
+use camino::{Utf8Path, Utf8PathBuf};
+use clap::Parser;
+use fs_err as fs;
+use sha2::Digest;
+use std::env;
+use std::io::{Read, Write};
+use walkdir::WalkDir;
+use xshell::Shell;
+
+/// This script will be placed in the live tests bundle to run the live tests
+const LIVE_TESTS_EXECUTION_SCRIPT: &str = r#"
+    set -euxo pipefail
+    /opt/oxide/omdb/bin/omdb -w nexus blueprints target enable current
+
+    TMPDIR=/var/tmp ./cargo-nextest nextest run \
+        --profile=live-tests \
+        --archive-file live-tests-archive/omicron-live-tests.tar.zst \
+        --workspace-remap live-tests-archive
+"#;
+
+#[derive(Parser)]
+pub struct A4x2PackageArgs {
+    /// Choose which source of oxidecomputer/testbed to build into the output.
+    #[clap(long, default_value_t = String::from("https://github.com/oxidecomputer/testbed"))]
+    testbed_source: String,
+
+    /// Choose which branch of oxidecomputer/testbed to build into the output.
+    #[clap(long, default_value_t = String::from("main"))]
+    testbed_branch: String,
+
+    /// Output package is written here.
+    #[clap(long, default_value_t = Utf8PathBuf::from(super::DEFAULT_A4X2_PKG_PATH))]
+    output: Utf8PathBuf,
+}
+
+struct Environment {
+    /// Path to `cargo` command
+    cargo: Utf8PathBuf,
+
+    /// Path to `git` command
+    git: Utf8PathBuf,
+
+    /// Directory from which xtask a4x2-package was invoked. Should be an
+    /// omicron source tree.
+    src_dir: Utf8PathBuf,
+
+    /// Directory within which we will build a4x2, omicron, tests, and then
+    /// package them
+    work_dir: Utf8PathBuf,
+
+    /// Path within `work_dir` where we will place build outputs, prior to
+    /// generation of the final tarball artifact.
+    out_dir: Utf8PathBuf,
+
+    /// Path within `work_dir` containing a copy of the omicron source tree
+    omicron_dir: Utf8PathBuf,
+}
+
+pub fn run_cmd(args: A4x2PackageArgs) -> Result<()> {
+    let sh = Shell::new()?;
+
+    let env = {
+        let cargo = Utf8PathBuf::from(
+            std::env::var("CARGO").unwrap_or(String::from("cargo")),
+        );
+        let git = Utf8PathBuf::from(
+            std::env::var("GIT").unwrap_or(String::from("git")),
+        );
+
+        let src_dir = Utf8PathBuf::try_from(env::current_dir()?)?;
+        let work_dir = src_dir.join("target/a4x2/package");
+
+        // Delete any results from previous runs.
+        if work_dir.try_exists()? {
+            fs::remove_dir_all(&work_dir)?;
+        }
+
+        // Create work dir
```

Comment:
same -- print a message saying it's doing this ("using working directory {work_dir}")